metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -583,9 +669,327 @@ def _get_datasource_urn(self, platform, platform_instance, database, source_tabl
+                else None
+            ),
+            env=self.config.env,
+            graph=self.ctx.graph,
+        )
+
+        schema_fields = infer_output_schema(parsed_query_object)
+        if schema_fields:
+
+            schema_metadata = SchemaMetadataClass(
+                schemaName="mode_query",
+                platform=f"urn:li:dataPlatform:{self.platform}",
+                version=0,
+                fields=schema_fields,
+                hash="",
+                platformSchema=OtherSchemaClass(rawSchema=""),
+            )
+            if self.config.tag_measures_and_dimensions:
+                self.set_field_tags(schema_fields)
+
+            yield (
+                MetadataChangeProposalWrapper(
+                    entityType="dataset",
+                    changeType=ChangeTypeClass.UPSERT,
+                    entityUrn=query_urn,
+                    aspectName="schemaMetadata",
+                    aspect=schema_metadata,
+                ).as_workunit()
+            )
+
+        yield from self.get_upstream_lineage_for_parsed_sql(
+            query_data, parsed_query_object
+        )
+
+        operation = OperationClass(
+            operationType="UPDATE",
+            lastUpdatedTimestamp=int(
+                dp.parse(query_data.get("updated_at", "now")).timestamp() * 1000
+            ),
+            timestampMillis=int(datetime.now(tz=timezone.utc).timestamp() * 1000),
+        )
+
+        yield MetadataChangeProposalWrapper(
+            entityType="dataset",
+            changeType=ChangeTypeClass.UPSERT,
+            entityUrn=query_urn,
+            aspectName="operation",
+            aspect=operation,
+        ).as_workunit()
+
+        creator = self._get_creator(
+            query_data.get("_links", {}).get("creator", {}).get("href", "")
+        )
+        modified_actor = builder.make_user_urn(
+            creator if creator is not None else "unknown"
+        )
+
+        created_ts = int(
+            dp.parse(query_data.get("created_at", "now")).timestamp() * 1000
+        )
+        modified_ts = int(
+            dp.parse(query_data.get("updated_at", "now")).timestamp() * 1000
+        )
+
+        query_instance_urn = self.get_query_instance_urn_from_query(query_data)
+        value = query_data.get("raw_query")
+        if value:
+            query_properties = QueryPropertiesClass(
+                statement=QueryStatementClass(
+                    value=value,
+                    language=QueryLanguageClass.SQL,
+                ),
+                source=QuerySourceClass.SYSTEM,
+                created=AuditStamp(time=created_ts, actor=modified_actor),
+                lastModified=AuditStamp(time=modified_ts, actor=modified_actor),
+            )
+
+            yield MetadataChangeProposalWrapper(
+                entityType="query",
+                changeType=ChangeTypeClass.UPSERT,
+                entityUrn=query_instance_urn,
+                aspectName="queryProperties",
```

Comment:
only entityUrn and aspect are required - everything else can be inferred automatically